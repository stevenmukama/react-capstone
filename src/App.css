* {
  font-family: 'Cabin', sans-serif;
}

body {
  margin: 0;
  padding: 0;
}

a {
  color: inherit;
  text-decoration: none;
}

.headerSlider {
  display: flex;
  align-items: center;
  gap: 1rem;
  background: #fd5294;
  background-repeat: no-repeat;
  background-size: cover;
  height: 25vh;
  padding: 2%;
  justify-content: center;
}

.headerSlider h4,
p {
  margin: 0;
}

.active-crypto {
  color: white;
}

.main-logo {
  width: 35%;
  opacity: 0.4;
}

.navbar {
  background: hsl(336, 81%, 61%);
  height: 3rem;
  display: flex;
  align-items: center;
  color: white;
  justify-content: space-between;
  padding: 1% 3%;
}

.coin-container {
  display: grid;
  grid-template-columns: 1fr 1fr;
}

.coin-link {
  background: #dc4782;
  padding: 7%;
  display: flex;
  justify-content: center;
  flex-direction: column;
  color: white;
  height: 20vh;
  border: 0;
  outline: 0;
  text-decoration: none;
}

.coin h3,
p {
  margin: 0;
}

.lastcoin {
  text-align: right;
}

.coin img {
  width: 7rem;
  margin: 5%;
}

.coin-link:nth-child(4n + 3),
.coin-link:nth-child(4n + 2) {
  background: #d1447a;
}

.menu {
  display: flex;
  gap: 1rem;
}

.cointop {
  height: 2.5rem;
  background: #d1447a;
  color: white;
  display: flex;
  align-items: center;
  padding-left: 2%;
  text-transform: uppercase;
  font-weight: 600;
}

.search-bar {
  background: #ec4c8b;
  display: flex;
  justify-content: center;
  margin-top: -2px;
}

.search {
  margin: 3% 0;
  width: 85%;
  height: 2.5rem;
  border: 0;
  outline: 0;
  padding: 0 4%;
  font-size: 1rem;
  border-radius: 50px;
  text-align: center;
}

@media only screen and (min-width: 768px) {
  .coin-container {
    grid-template-columns: 1fr 1fr 1fr 1fr;
  }

  .coin img {
    width: 8rem;
  }

  .menu {
    gap: 2rem;
  }

  .lastcoin {
    text-align: center;
  }

  .cointop {
    height: 4rem;
    font-size: 1.5rem;
  }

  .headerSlider {
    height: 35vh;
    gap: 4rem;
  }

  .main-logo {
    width: 20%;
  }

  .coin h3 {
    font-size: 1.5rem;
  }

  .coin p {
    font-size: 1.2rem;
  }

  .coin-link {
    height: 25vh;
    align-items: center;
    text-align: center;
    border: 0.5px solid rgb(74, 56, 61);
    background: #d1447a;
  }

  .coin-link:nth-child(even) {
    background: #dc4782;
  }

  .headerSlider h4 {
    font-size: 2.5rem;
  }

  .headerSlider p {
    font-size: 2rem;
  }

  .navbar {
    height: 5rem;
    font-size: 1.5rem;
  }

  .search {
    width: 40%;
    height: 3rem;
    margin: 0;
  }

  .search-bar {
    height: 4rem;
  }
}
