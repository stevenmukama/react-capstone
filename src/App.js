import './App.css';
import { Routes, Route, BrowserRouter } from 'react-router-dom';
import Home from './pages/Home';
import Navbar from './components/Navbar';
import Details from './pages/Details';

function App() {
  return (
    <div className="App">
      <BrowserRouter>
        <Navbar />
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="/details" element={<Details />}>
            <Route path=":coinId" element={<Details />} />
          </Route>
        </Routes>
      </BrowserRouter>
    </div>
  );
}

export default App;
