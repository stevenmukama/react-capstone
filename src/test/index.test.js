import React from 'react';
import ReactDOM from 'react-dom/client';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import App from '../App';
import store from '../redux/configureStore';

describe('App', () => {
  test('renders without crashing', () => {
    const div = document.createElement('div');
    div.id = 'root';
    document.body.appendChild(div);

    const root = ReactDOM.createRoot(div);
    root.render(
      <React.StrictMode>
        <Provider store={store}>
          <App />
        </Provider>
      </React.StrictMode>,
    );
    root.unmount();
  });

  test('matches snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
