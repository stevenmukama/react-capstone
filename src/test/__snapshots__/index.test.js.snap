// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<div
  className="App"
>
  <div
    className="navbar"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <i
        className="fa-solid fa-chevron-left"
      />
    </a>
    <p>
      Crypto Metrics
    </p>
    <div
      className="menu"
    >
      <i
        className="fa-solid fa-microphone"
      />
      <i
        className="fa-solid fa-gear"
      />
    </div>
  </div>
  <div
    className="search-bar"
  >
    <input
      className="search"
      onChange={[Function]}
      placeholder="Search for Cryptocurrency"
      type="text"
      value=""
    />
  </div>
  <div
    className="headerSlider"
  >
    <img
      alt="crypto"
      className="main-logo"
      src="https://upload.wikimedia.org/wikipedia/commons/thumb/8/8b/Cryptocurrency_Logo.svg/633px-Cryptocurrency_Logo.svg.png"
    />
    <div
      className="active-crypto"
    >
      <h4>
        ACTIVE
      </h4>
      <p>
         
        currencies
      </p>
    </div>
  </div>
  <div
    className="cointop"
  >
    <p>
      Cryptocurrencies
    </p>
  </div>
  <div
    className="coin-container"
  />
</div>
`;
